release-23.1: bump version to v23.1.17

celiala opened the pull request:
Release note: None
Epic: None
Release justification: non-production (release infra) change.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/119551)
<!-- Reviewable:end -->

celiala committed:
release-23.1: bump version to v23.1.17

Release note: None
Epic: None
Release justification: non-production (release infra) change.
